userspace/libscap/examples/01-open/scap_open.c

```diff
@@ -53,15 +55,110 @@ enum scap_source
 /* Configuration variables set through CLI. */
 int source = -1;		  /* scap source to catch events. */
 uint64_t num_events = UINT64_MAX; /* max number of events to catch. */
-bool simple_consumer = false;	  /* kernel simple consumer mode. */
 int evt_type = -1;		  /* event type to print. */
+interesting_tp_set tp_of_interest;
+bool tp_is_set = false;
+interesting_ppm_sc_set ppm_sc_of_interest;
+bool ppm_sc_is_set = false;
 
 /* Generic global variables. */
 scap_open_args args = {.mode = SCAP_MODE_LIVE}; /* scap args used in `scap_open`. */
 uint64_t g_nevts = 0;				/* total number of events captured. */
 scap_t* g_h = NULL;				/* global scap handler. */
 uint16_t* lens16 = NULL;			/* pointer used to print the length of event params. */
 char* valptr = NULL;				/* pointer used to print the value of event params. */
+struct timeval tval_start, tval_end, tval_result;
+
+void enable_single_tp(const char* tp_basename)
+{
+	bool found = false;
+
+	for(int i = 0; i < TP_VAL_MAX && !found; i++)
+	{
+		if(strcmp(tp_names[i], tp_basename) == 0)
+		{
+			tp_of_interest.tp[i] = true;
+			found = true;
+		}
+	}
+
+	if(!found)
+	{
+		fprintf(stderr, "Tracepoint '%s' not found. Unsupported or wrong parameter?\n", tp_basename);
+		fprintf(stderr, "Please choose between:\n");
+		for(int i = 0; i < TP_VAL_MAX; i++)
+		{
+			fprintf(stderr, "\t* %s\n", tp_names[i]);
+		}
+		exit(EXIT_FAILURE);
+	}
+	else
+	{
+		tp_is_set = true;
+	}
+}
+
+void enable_single_ppm_sc(int ppm_sc_code)
+{
+	if(ppm_sc_code == -1)
+	{
+		/* In this case we won't have any syscall enabled. */
+		ppm_sc_is_set = true;
+		return;
+	}
+
+	if(ppm_sc_code < 0 || ppm_sc_code >= PPM_SC_MAX)
+	{
+		fprintf(stderr, "Unexistent ppm_sc code: %d. Wrong parameter?\n", ppm_sc_code);
+		exit(EXIT_FAILURE);
+	}
+	ppm_sc_of_interest.ppm_sc[ppm_sc_code] = true;
+	ppm_sc_is_set = true;
+}
+
+void set_enabled_syscalls()
+{
+	printf("---------------------- INTERESTING SYSCALLS ----------------------\n");
+	if(ppm_sc_is_set)
+	{
+		args.ppm_sc_of_interest = &ppm_sc_of_interest;
+		printf("* Syscall enabled:\n");
+		for(int j = 0; j < PPM_SC_MAX; j++)
+		{
+			if(args.ppm_sc_of_interest->ppm_sc[j])
+			{
+				printf("- %s\n", g_syscall_info_table[j].name);
```

Comment:
This PR is 🚀 . In addition, these `scap-open example` extensions are amazing for easy performance studies. Truly thanks a lot for adding all these new options!  @FedeDP @Andreagit97 

We already have the `--print_syscalls` arg. Have found it a bit cumbersome to look up the right `--ppm_sc` code for the system call intended to be enabled. Could printing an index like `syscall_name: <NAME> syscall_nr: <SYSCALL_NR> ppm_sc: <PPM_SC_CODE_NR>`  be something that could be supported?

Reply:
Done :)